### PR TITLE
net/mwan3: fix empty gateway when creating routing table

### PR DESCRIPTION
**Pull request by yangfl**
Maintainer: @feckert
Compile tested: arch:ar71xx, OpenWRT/LEDE version:Reboot 17.01.4, tests done
Run tested: arch:ar71xx, OpenWRT/LEDE version:Reboot 17.01.4, tests done

Description:
Interfaces of PtP protocol do not have a real gateway, and ubus fills these fields with '0.0.0.0'. This will cause error when adding new routing rule.
Signed-off-by: David Yang <mmyangfl@gmail.com>

### CODE REVIEW

File: net/mwan3/files/lib/mwan3/mwan3.sh [hunk 1]
```diff
@@ -322,32 +322,37 @@ mwan3_create_iface_route()
...
 		$IP4 route flush table $id
-		$IP4 route add table $id default $route_args
+		$IP4 route add table $id default $route_args dev $2
 	fi
 
 	if [ "$family" == "ipv6" ]; then
-
-		ubus call network.interface.${1}_6 status &>/dev/null
-		if [ "$?" -eq "0" ]; then
+		if ubus call network.interface.${1}_6 status &>/dev/null; then
```
Comment (feckert) [inline net/mwan3/files/lib/mwan3/mwan3.sh]: Could you do this please in a separate commit?
This is not reflected in the commit message.
And has nothing to do with this commit.

File: net/mwan3/files/lib/mwan3/mwan3.sh [hunk 2]
```diff
@@ -322,32 +322,37 @@ mwan3_create_iface_route()
 	[ -n "$id" ] || return 0
 
 	if [ "$family" == "ipv4" ]; then
-		ubus call network.interface.${1}_4 status &>/dev/null
-		if [ "$?" -eq "0" ]; then
```
Comment (feckert) [inline net/mwan3/files/lib/mwan3/mwan3.sh]: Could you do this please in a separate commit?
This is not reflected in the commit message.
And has nothing to do with this commit.

**Comment by feckert** [comment]
@yangfl i have no PtP protocol in use.
Will have look in the next days if this could be applied.
Thanks for your contribution :-)

File: net/mwan3/files/lib/mwan3/mwan3.sh (hunk 2 above)
Comment (yangfl) [reply net/mwan3/files/lib/mwan3/mwan3.sh]: It simply does exactly the same thing as the origin code and it reduces one extra comparison.
I've seen many of such superfluous usage in the code. If you don't like I can just undo it. :(

**Comment by yangfl** [comment]
Ok, reordered.

File: net/mwan3/files/etc/hotplug.d/iface/15-mwan3 [hunk 3]
```diff
@@ -26,24 +26,20 @@ if [ "$ACTION" == "ifup" ]; then
 	if [ "$family" = "ipv4" ]; then
 		ubus call network.interface.${INTERFACE}_4 status &>/dev/null
 		if [ "$?" -eq "0" ]; then
-			network_get_gateway gateway ${INTERFACE}_4
```
Comment (feckert) [inline net/mwan3/files/etc/hotplug.d/iface/15-mwan3]: Why do you want to delete the gateway check on this interface? Could you pleas explain this?

Comment (yangfl) [reply net/mwan3/files/etc/hotplug.d/iface/15-mwan3]: Since https://github.com/openwrt/packages/pull/5033/files#diff-ccaf6dffccf2fc134d63b5c33d5ee8e1R327 . gateway is no longer used anymore.

Comment (feckert) [reply net/mwan3/files/etc/hotplug.d/iface/15-mwan3]: sorry i dont get it ?

Comment (yangfl) [reply net/mwan3/files/etc/hotplug.d/iface/15-mwan3]: In such protocol (yes, OpenVPN included), no gateway are set. Example:
`default dev tun0 proto static scope link metric 1000`
So uci may fill them '0.0.0.0', or just empty string in some cases.
This won't affect currently runnable config.

Comment (feckert) [reply net/mwan3/files/etc/hotplug.d/iface/15-mwan3]: @yangfl
Thanks for clarification i will have a look in the next days on this topic. Maybe on Friday.

**Comment by feckert** [comment]
@yangfl could you please update this commit?
https://github.com/openwrt/packages/pull/5033/commits/5a66eb58154035b8cffa9f705427431ef60aaf03

travis-ci/pr is not satisfied!
> 
> === Checking commit '5a66eb58154035b8cffa9f705427431ef60aaf03'
> 
> Author name (yangfl) need to be your real name 'firstname lastname'
> 
> Commit subject line seems ok (mwan3: add check for ping)
> 
> Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: yangfl <yangfl@users.noreply.github.com>')

**Comment by yangfl** [comment]
Done @feckert

**Comment by feckert** [comment]
@yangfl LGTM
I have tested your changes with the following proto-handllers "pppoe, static, dhcp and qmi".
And my setup works as before. Load balancing and failover.

Do you made tests with 3rd-party scripts and PtP proto-handler which did not worked before with mwan3?

Could you please also update the mwan3 version 2.6.7 with this PR?

**Comment by yangfl** [comment]
@feckert These changes are especially designed for solving problems with OpenVPN, i.e. https://github.com/openwrt/packages/issues/3486, and other PtPs (gre, ipip, 6in4, and more).

**Comment by feckert** [comment]
@yangfl could you also please update the version to 2.6.7?

**Comment by yangfl** [comment]
@feckert I did. See commit above.

**Comment by feckert** [comment]
@yangfl sorry for get my last comment. I had an browser issue!!
